src/App.tsx

```diff
@@ -1,61 +1,129 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { User, Todo } from './types/types';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+function getUserById(userId: number): User | null {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+export const todos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [selectedUserId, setselectedUserId] = useState(0);
+  const [title, setTitle] = useState('');
+  const [currentTodos, setCurrentTodos] = useState(todos);
+  const [hasUserError, setHasUserError] = useState(false);
+  const [hasTitleError, setHasTitleError] = useState(false);
+
+  function getNewTodo() {
+    const newId = Math.max(...currentTodos.map(todo => todo.id)) + 1;
+
+    return {
+      id: newId,
+      title,
+      completed: false,
+      userId: selectedUserId,
+      user: getUserById(selectedUserId),
+    };
+  }
+
+  const titleHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const selectedUserHandler = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setselectedUserId(+event.target.value);
+    setHasUserError(false);
+  };
+
+  const buttonHandler = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserError(!selectedUserId);
+
+    if (title && selectedUserId) {
+      setCurrentTodos([
+        ...currentTodos,
+        getNewTodo(),
```

Comment:
this should be stored into a separate variable instead of calling it on submit. What's more, from its name I guess that it somewhere gets the todo and it is not about input.